File: src/main/java/org/padler/gradle/minify/minifier/Event.java
package org.padler.gradle.minify.minifier;

import com.google.common.css.compiler.ast.GssError;
import com.google.javascript.jscomp.JSError;

public class Event {
    private int charNo;
    private int lineNo;
    private String message;
    private String sourceFileName;

    public Event(GssError error) {
        charNo = error.getLocation().getBeginCharacterIndex();
        lineNo = error.getLocation().getLineNumber();
        message = error.getMessage();
        sourceFileName = error.getLocation().getSourceCode().getFileName();
    }

    public Event(JSError error) {
        charNo = error.getCharno();
        lineNo = error.getLineNumber();
        message = error.getDescription();
        sourceFileName = error.getSourceName();
    }

    @Override
    public final String toString() {
        return message
                + " at "
                + sourceFileName
                + " line "
                + lineNo
                + " : "
                + charNo;
    }
}

File: src/main/kotlin/org/padler/gradle/minify/minifier/result/Event.kt
package org.padler.gradle.minify.minifier.result

import com.google.common.css.compiler.ast.GssError
import com.google.javascript.jscomp.JSError

open class Event {

    private var charNo: Int
    private var lineNo: Int
    private var message: String
    private var sourceFileName: String

    constructor(error: GssError) {
        charNo = error.location.beginCharacterIndex
        lineNo = error.location.lineNumber
        message = error.message
        sourceFileName = error.location.sourceCode.fileName
    }

    constructor(error: JSError) {
        charNo = error.charno
        lineNo = error.lineNumber
        message = error.description
        sourceFileName = error.sourceName
    }

    override fun toString(): String {
        return "$message at $sourceFileName line $lineNo : $charNo"
    }
}
